controlplane/eks/api/v1beta2/awsmanagedcontrolplane_types.go

```diff
@@ -173,6 +173,10 @@ type AWSManagedControlPlaneSpec struct { //nolint: maligned
 	// +optional
 	VpcCni VpcCni `json:"vpcCni,omitempty"`
 
+	// FilterPrivateSubnets indicates that the Amazon EKS control plane should only use private subnets.
+	// +optional
+	FilterPrivateSubnets bool `json:"filterPrivateSubnets,omitempty"`
```

Comment:
I'm wonder if we should be looking at the `Private` field on the endpoint access: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/2a28b831208b17473e2ec9a1c63928d379c03b05/controlplane/eks/api/v1beta2/awsmanagedcontrolplane_types.go#L219

Can't decide if that should be enough.

Reply:
I wanted to keep this flag separate from the endpoint access.

Reply:
Ok cool 👍 Could we use a name that doesn't include "Filter"? Using "Filter" feels like it's exposing what we are doing internally instead of what we want, which to make the CP attach to private subnets.